crates/matrix-sdk-store-encryption/src/lib.rs

```diff
@@ -284,16 +284,83 @@ impl StoreCipher {
     /// # Result::<_, anyhow::Error>::Ok(()) };
     /// ```
     pub fn encrypt_value(&self, value: &impl Serialize) -> Result<Vec<u8>, Error> {
-        let mut data = serde_json::to_vec(value)?;
+        Ok(serde_json::to_vec(&self.encrypt_value_typed(value)?)?)
+    }
+
+    /// Encrypt a value before it is inserted into the key/value store.
+    ///
+    /// A value can be decrypted using the
+    /// [`StoreCipher::decrypt_value_typed()`] method. This is the lower
+    /// level function to `encrypt_value`, but returns the
+    /// full `EncryptdValue`-type
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - A value that should be encrypted, any value that implements
+    /// `Serialize` can be given to this method. The value will be serialized as
+    /// json before it is encrypted.
+    ///
+    ///
+    /// # Examples
+    ///
+    /// ```
```

Comment:
Since this doesn't do anything anyways, let's stop `cargo test` from spawning an unnecessary process:

```suggestion
    /// ```no_run
```